File: bluetooth/src/main/java/com/ankerboxmanager/bluetooth/scaner/NormalBluetoothScanner.java
//package com.ankerboxmanager.bluetooth.scaner;
//
//import com.ankerboxmanager.bluetooth.IBluetooth;
//
//import android.bluetooth.BluetoothAdapter;
//import android.bluetooth.BluetoothDevice;
//import android.content.BroadcastReceiver;
//import android.content.Context;
//import android.content.Intent;
//import android.content.IntentFilter;
//
///**
// * Created on 15/05/2018.
// *
// * @author lx
// */
//
//public class NormalBluetoothScanner extends BluetoothScanner {
//
//    private BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
//        @Override
//        public void onReceive(Context context, Intent intent) {
//            if (intent == null || intent.getAction() == null) {
//                return;
//            }
//            if (BluetoothDevice.ACTION_FOUND.equals(intent.getAction())) {
//                BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
//                onNewDeviceFound(device);
//            }
//        }
//    };
//
//    public NormalBluetoothScanner(Context context) {
//        super(context);
//    }
//
//    @Override
//    public void startScan(String[] serviceUUIDs, IBluetooth.AsyncReturnCallback asyncReturnCallback) {
//        context.registerReceiver(broadcastReceiver, new IntentFilter(BluetoothDevice.ACTION_FOUND));
//        boolean result =  getBluetoothAdapter().startDiscovery();
//        if (asyncReturnCallback != null) {
//            asyncReturnCallback.onReturn(result);
//        }
//    }
//
//    @Override
//    public void stopScan(IBluetooth.AsyncReturnCallback asyncReturnCallback) {
//        context.unregisterReceiver(broadcastReceiver);
//        boolean result =  getBluetoothAdapter().cancelDiscovery();
//        if (asyncReturnCallback != null) {
//            asyncReturnCallback.onReturn(result);
//        }
//    }
//
//    @Override
//    public void destroy() {
//        try {
//            context.unregisterReceiver(broadcastReceiver);
//        } catch (Exception e) {
//        }
//        super.destroy();
//    }
//
//    private BluetoothAdapter getBluetoothAdapter() {
//        return BluetoothAdapter.getDefaultAdapter();
//    }
//
//}

File: bluetooth/src/main/kotlin/com/ankerboxmanger/bluetooth/scaner/NormalBluetoothScanner.kt
package com.ankerboxmanger.bluetooth.scaner

import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import com.ankerboxmanger.bluetooth.IBluetooth

/**
 * Created on 15/05/2018.
 *
 * @author lx
 */

class NormalBluetoothScanner(context: Context) : BluetoothScanner(context) {

    private val broadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent?) {
            if (BluetoothDevice.ACTION_FOUND == intent?.action) {
                val device = intent.getParcelableExtra<BluetoothDevice>(BluetoothDevice.EXTRA_DEVICE)
                onNewDeviceFound(device)
            }
        }
    }

    private val bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()

    override fun startScan(serviceUUIDs: Array<String>?, asyncReturnCallback: IBluetooth.AsyncReturnCallback?) {
        context.registerReceiver(broadcastReceiver, IntentFilter(BluetoothDevice.ACTION_FOUND))
        val result = bluetoothAdapter.startDiscovery()
        asyncReturnCallback?.onReturn(result)
    }

    override fun stopScan(asyncReturnCallback: IBluetooth.AsyncReturnCallback?) {
        context.unregisterReceiver(broadcastReceiver)
        val result = bluetoothAdapter.cancelDiscovery()
        asyncReturnCallback?.onReturn(result)
    }

    override fun destroy() {
        try {
            context.unregisterReceiver(broadcastReceiver)
        } catch (e: Exception) {
        }
        super.destroy()
    }

}
